catalog/ui/search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/controller/search/QueryRunnable.java

```diff
@@ -255,22 +254,13 @@ protected QueryResponse queryCatalog(String sourceId, SearchRequest searchReques
                 }
 
                 LOGGER.debug("Sending query: {}", query);
-                response = searchController.getFramework()
-                        .query(request);
+                ResultIterable results = new ResultIterable(searchController.getFramework(),
+                        request);
+                List<Result> resultsList = new ArrayList<>();
+                results.iterator()
```

Comment:
As soon as https://github.com/codice/ddf/pull/2207 is merged, there will be a way to cap the maximum number of results returned by the ResultIterable.